### ISSUE TITLE
Contents of file are read multiple times given the progression of position

### ISSUE THREAD

**Issue by silkentrance** [neurosnap/gen-readlines#3]
In case of foundNewLine >= 0, the progression of position leads to multiple reads of the same data from the input file.

See line

https://github.com/neurosnap/gen-readlines/blob/master/index.js#L40

which will just add the length of the newline buffer, subsequently causing the rest of the read buffer to be discarded and read again in the next iteration.

Here is what I came up with to solve that problem

```
 35    let foundNewline = _foundNewline(readChunk);
 36     while (foundNewline >= 0)
 37     {
 38         let newlineBuffer = new Buffer(readChunk.slice(0, foundNewline));
 40         yield _concat(lineBuffer, newlineBuffer);
 42         readChunk = readChunk.slice(foundNewline+1);
 43         foundNewline = _foundNewline(readChunk);
 44     }
 45     position += bufferSize;
 46     lineBuffer = _concat(lineBuffer, readChunk);
 47  }
 48  // dump whatever...
```

I already applied your solution to the missing empty lines here.

**Comment by silkentrance** [comment]
This might be further optimized by keeping a second inreadbuffer position pointer so that the second slice on the readchunk can be omitted.

And is new Buffer(readChunk.slice(...)) not redundant as well, given that slice() will return a new buffer?

**Comment by neurosnap** [comment]
Yes you're right, this might be why I avoided preserving the newlines in the output.  Can you submit a PR with these changes?

Thanks!

**Comment by silkentrance** [comment]
Will do, but I think I need to also fix #4 as well and provide a PR for both.

**Comment by neurosnap** [comment]
Any help is greatly appreciated!
